index.js

```diff
@@ -0,0 +1,39 @@
+const validateLineup = (lineup) => {
+  function getSalaries(lineup) {
+    return lineup.salary
+  }
+  function getSalarySum(acc, n) {
+    return acc + n
+  }
+  function getTeamId(lineup) {
+    return lineup.teamId
+  }
+  function getGameId(lineup) {
+    return lineup.gameId
+  }
+  function getPositions(lineup) {
+    return lineup.position
+  }
+
+  const salaryList = lineup.map(getSalaries)
+  const salarySum = salaryList.reduce(getSalarySum, 0)
+
+  const teamIdList = lineup.map(getTeamId)
+  const findTeamDuplicates = teamIdList.filter((item, index) => teamIdList.indexOf(item) != index)
```

Comment:
This isn't really doing what you think it is doing. There are 9 players in a lineup. If you had the first 8 come from 4 teams you'd end up with a teamIdList something like `[1,1,2,2,3,3,4,4,5]`. This would lead to a `findTeamDuplicates` of `[1,2,3,4]`. Which has a length of 4 thus violating your check on line 32 even though it is a valid lineup because you don't have "more than 2 players from a single team"